src/main/java/com/example/undersstandingmapping/controllers/StudentController.java

```diff
@@ -3,29 +3,51 @@
 import com.example.undersstandingmapping.models.Student;
 import com.example.undersstandingmapping.repositories.StudentRepository;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.*;
 
 import java.util.List;
 
 // to mark this class as a controller.
 // and return the output as json directly
 @RestController
 public class StudentController {
-
+    @Autowired
+    Student student;
```

Comment:
Why autowire student class ?

Reply:
Sir, I made Student class as @Component.. i have removed it and its working, i cant fetch and see the single student by search the name, Thanks so much. But no idea why the @autowired stopped this